practice-app/api/views.py

```diff
@@ -49,63 +49,122 @@ def post(self, *args, **kwargs):
         return Response(serializer.data, status=status.HTTP_200_OK)
 
 
+class InvestmentProfitAPIView(APIView):
+    authentication_classes = (JSONWebTokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    # Helper function for calculating profit of an investment with respect to given currency.
+    # Default is Turkish Liras.
+
+    @staticmethod
+    def _get_profit(investment, symbol="TRY"):
+        default_eq = Equipment.objects.get(symbol=symbol)
+        base_eq = investment.base_equipment
+        target_eq = investment.target_equipment
+        base_amount = investment.base_amount
+        target_amount = investment.target_amount
+
+        # If the sold equipment is the same as default equipment, ratio is 1.
+        if base_eq.symbol == default_eq.symbol:
+            base_default = 1
+        else:
+            base_default = Parity.objects.order_by('-date').filter(base_equipment=base_eq,
+                                                                   target_equipment=default_eq)[0].ratio
+        # If the bought equipment is the same as default equipment, ratio is 1.
+        if target_eq.symbol == default_eq.symbol:
+            target_default = 1
+        else:
+            target_default = Parity.objects.order_by('-date').filter(base_equipment=target_eq,
+                                                                     target_equipment=default_eq)[0].ratio
+        current = target_amount * target_default
+        would_be = base_amount * base_default
+        profit = float(current - would_be)
+
+        return profit
+
+    def post(self, request):
+        user_id = request.user.id
+        user = User.objects.get(id=user_id)
+        symbol = request.POST.get('symbol')
+
+        # If no symbol is given, use TRY as default.
+
+        if symbol is None:
+            symbol = "TRY"
+
+        # If no such investment exists in the DB, returning 400.
+        else:
+            symbols = Equipment.objects.values("symbol")
+            symbols = [obj["symbol"] for obj in symbols]
+
```

Comment:
Instead of fetching all of the symbols u could do a filter(symbol=symbol).exists() . You dont need all of the symbols if i am not mistaken.

Reply:
right, fixing now